### DISCUSSION TITLE
How to run the LLVM IR which is dumped by wasmedgec in LLVM?

### DISCUSSION THREAD

**Discussion by hungryzzz** [WasmEdge/WasmEdge#2391] [Q&A] [answered]
Hi, I try to run the LLVM IR which is dumped by wasmedgec in LLVM, but it needs external symbols **_"intrinsics"_**.

So what are the **_"intrinsics"_** symbols? Are they some runtime functions? Or can I link such **_"intrinsics"_** symbols with the LLVM IR?

Thank you!

**Comment by alabulei1** [comment]
@hydai Could you please take a look?

**Comment by hydai** [comment, marked as answer]
Hi, the intrinsics table is some host functions in WasmEdge: https://github.com/WasmEdge/WasmEdge/blob/dad018af52781d4344e7e1bbf2fd4771d49e6dd8/lib/executor/engine/proxy.cpp#L45-L70

**Reply by hungryzzz** [reply]
Hi, I got it, but I am wondering that why some Wasm bytecode should be implemented by such intrinsics and not translated to LLVM IR. 
Because if I use Emscripten to translate a C program to Wasm, the Emscripten uses LLVM to do it, which means the C program will be translated to LLVM IR then to the Wasm. So the Wasm bytecode should be fully retranslated to LLVM IR.
Is it related to some security issues?

**Reply by hydai** [reply]
For the security issue, we don't allow the AoT-generated binary to handle the memory. All related functions must use intrinsic proxy to interact with wasmedge internal functions.
Also, please keep in mind that the generated AoT binaries are not standalone executables; they will require wasmedge runtime to execute.

**Reply by hungryzzz** [reply]
Got it! Thank you very much!
